src/main/java/com/google/step/snippet/servlets/UserSettingServlet.java

```diff
@@ -24,11 +32,68 @@
 /** Servlet that renders user setting page */
 @WebServlet("/user")
 public class UserSettingServlet extends HttpServlet {
+  private static final String WEBSITE_PARAMETER = "website";
+  private static final String LANGUAGE_PARAMETER = "language";
+  private static final String ID_PARAMETER = "id";
+  private static final String EMAIL_PARAMETER = "email";
+  private static final String USER_CLASS_PARAMETER = "UserInfo";
+  private static final String USER_PARAMETER = "user";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-    // Forward the request to the template (which is a servlet itself).
+    UserService userService = UserServiceFactory.getUserService();
+    // In case unauthenticated user access setting page through URL.
+    if (!userService.isUserLoggedIn()) {
+      doPost(request, response);
+      return;
+    }
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity userEntity = getUserEntity(datastore, userService);
+    String website = null;
+    String language = null;
+    if (userEntity != null) {
+      website = (String) userEntity.getProperty(WEBSITE_PARAMETER);
+      language = (String) userEntity.getProperty(LANGUAGE_PARAMETER);
+    }
+    User user =
+        new User(
+            (String) userEntity.getProperty(ID_PARAMETER),
+            (String) userEntity.getProperty(EMAIL_PARAMETER),
+            website,
+            language);
+    request.setAttribute(USER_PARAMETER, user);
     request.getRequestDispatcher("WEB-INF/templates/user_dashboard.jsp").forward(request, response);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/");
+      return;
+    }
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String website = request.getParameter(WEBSITE_PARAMETER);
```

Comment:
You should probably validate that website is one of GeeksForGeeks, StackOverflow, etc.